packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -564,6 +564,13 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
       if (!_visible) {
         return;
       }
+      // Check if there's an ongoing route transition
+      final ModalRoute<dynamic>? route = ModalRoute.of(context);
+      if (route?.secondaryAnimation != null && route!.secondaryAnimation!.isAnimating) {
+        _timer?.cancel();
+        _timer = Timer(route.transitionDuration, show);
```

Comment:
Is it possible to listen to when the animation ends and start the timer then? The current way isn't compatible with cases where this happens halfway during the transition or simulation-based transition where there isn't a predefined duration.

Reply:
Yaa, was thinking the same. but for some reason below written code doesn't fix the issue.
```dart
if (route?.secondaryAnimation != null && route!.secondaryAnimation!.isAnimating) {
   _timer?.cancel();
   void onAnimationDismiss(AnimationStatus status) {
        if (status == AnimationStatus.dismissed) {
           route.secondaryAnimation!.removeStatusListener(onAnimationDismiss);
           show();
       }
   }

    route.secondaryAnimation!.addStatusListener(onAnimationDismiss);
    return;
}
```

Reply:
In some cases the secondaryAnimation is swapped for a different one, so that may be why. The secondaryAnimation with the status listener is replaced.

Reply:
Rather than putting the logic here, should we have the check happen in tandem with the `_visible` check for `_ExclusiveMouseRegion`? So it ignores all mouse events while the route transitions are in motion?

Reply:
Pushed the change, everything looks good. Thanks.